nixos/modules/services/mail/postsrsd.nix

```
@@ -7,6 +7,23 @@
 let
 
   cfg = config.services.postsrsd;
+  runtimeDirectoryName = "postsrsd";
+  runtimeDirectory = "/run/${runtimeDirectoryName}";
+  # TODO: follow RFC 42, but we need a libconfuse format first:
+  #       https://github.com/NixOS/nixpkgs/issues/401565
+  # Arrays in `libconfuse` look like this: {"Life", "Universe", "Everything"}
+  # See https://www.nongnu.org/confuse/tutorial-html/ar01s03.html
+  libconfuseDomains = "{ " + lib.concatStringsSep ", " (map (x: ''"${x}"'') cfg.domains) + " }";
```

Would it make sense to use something like `toJSON` to serialise the string values? This would handle chars that need escaping (e.g. serialising `"` as `\"`).

```suggestion
  libconfuseDomains = "{ " + lib.concatStringsSep ", " (map builtins toJSON) cfg.domains) + " }";
```

Also: you can use [`lib.concatMapStringsSep`](https://noogle.dev/f/lib/concatMapStringsSep) to combine `map` and `concatStringsSep` into a single call:

```suggestion
  libconfuseDomains = "{ " + lib.concatMapStringsSep ", " builtins.toJSON cfg.domains + " }";
```

---

Oh, nice! For the record, this isn't exactly correct. This json expression represents a 4 character string: `"${F}"`, whereas in libconfig, that actually evaluates to some env var. The way to represent this in libconfig is either `'${F}'` or `"\${F}"`.

That said, it's not *worse* than what I was doing, so I've made this change.

---

If it's just a case of additional escaping, you may be able to use [`lib.strings.escape`](https://noogle.dev/f/lib/strings/escape) either before or after `toJSON`?